esx_service/utils/auth_api.py

```diff
@@ -631,10 +637,20 @@ def _tenant_access_set(name, datastore, allow_create=None, volume_maxsize_in_MB=
         error_info = ErrorInfo(err_code, err_msg)
         return error_info
 
+    allow_create_val = str(allow_create)
+    allow_create_val = allow_create_val.lower()
```

Comment:
Yes this looks better. Strings can be converted to bools; but the end value of allow_create needs to be True, False or None ( and not just True or False) to distinguish between set privilege, unset privilege and don't change privilege.